graphistry/layout/ring/time.py

```diff
@@ -0,0 +1,391 @@
+from typing import Any, Callable, Dict, List, Literal, Optional, Tuple, Union
+import numpy as np
+import pandas as pd
+from functools import lru_cache
+from graphistry.Plottable import Plottable
+
+
+TimeUnit = Literal['s', 'm', 'h', 'D', 'W', 'M', 'Y', 'C']
+"""Time unit for axis labels
+
+- 's': seconds
+- 'm': minutes
+- 'h': hours
+- 'D': days
+- 'W': weeks
+- 'M': months
+- 'Y': years
+- 'C': centuries
+
+"""
+
+
+@lru_cache
+def unit_to_timedelta(unit: TimeUnit) -> np.timedelta64:
+    map: Dict[TimeUnit, np.timedelta64] = {
+        's': np.timedelta64(1, 's'),
+        'm': np.timedelta64(60, 's'),
+        'h': np.timedelta64(3600, 's'),
+        'D': np.timedelta64(86400, 's'),
+        'W': np.timedelta64(7 * 86400, 's'),
```

Comment:
Maybe we can put some of these computed numbers as constants:
`WEEK = 604800  # 7 * 86400`